### DISCUSSION TITLE
Updating temperature values within the experiment!

### DISCUSSION THREAD

**Discussion by mlnsai** [pybamm-team/PyBaMM#2070] [Q&A] [answered]
Hey Team!

I am carrying out a typical charge discharge experiment for a li-ion battery for 24 HOURS . And for that I need to update ambient temperature EVERY HOUR. I was wondering whether there is a way to update the temperature while the experiment is going on for every hour. Please do let me know

**Comment by TomTranter** [comment, marked as answer]
Yeah you could make the parameter a function and interpolate or you could make it an input and step the simulation and provide new values on the fly. Have a look through the examples to find ways of doing each.

**Reply by mi3618P** [reply]
Hi! I was wondering if the step method could be used to update a spatially variable ambient temperature? How should the InputParameter be set up in that case? Making the parameter a function in the first place using the conventioanl method in the [PouchCellCooling](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/pouch_cell_cooling.py) example would not be suitable for the coupled problem I am trying to solve, as the ambient temperature would not be known a priori. Thank you.

**Comment by mlnsai** [comment]
Thanks Tom for the timely response. This helped me. Cheers :)
